src/Nethermind/Nethermind.Evm.Test/Eip7667Tests.cs

```diff
@@ -0,0 +1,156 @@
+using Nethermind.Core.Specs;
+using Nethermind.Evm.Precompiles;
+using Nethermind.Int256;
+using Nethermind.Specs.Forks;
+using Nethermind.Specs.Test;
+using NUnit.Framework;
+
+namespace Nethermind.Evm.Test;
+
+public class Eip7667Tests : VirtualMachineTestsBase
+{
+    [SetUp]
+    public void SetUp()
+    {
+        Setup();
+    }
+
+    protected override ISpecProvider SpecProvider => new CustomSpecProvider(
+        ((ForkActivation)0, Frontier.Instance),
+        ((ForkActivation)1, Cancun.Instance),
+        ((ForkActivation)2, Eip7667Spec.Instance));
+
+    [Test]
+    public void GetSha3Cost()
+    {
+        Assert.That(Cancun.Instance.GetSha3Cost(), Is.EqualTo(30));
+        Assert.That(Eip7667Spec.Instance.GetSha3Cost(), Is.EqualTo(300));
+    }
+
+    [Test]
+    public void GetSha3WordCost()
+    {
+        Assert.That(Cancun.Instance.GetSha3WordCost(), Is.EqualTo(6));
+        Assert.That(Eip7667Spec.Instance.GetSha3WordCost(), Is.EqualTo(60));
+    }
+
+    [Test]
+    public void GetLogDataCost()
+    {
+        Assert.That(Cancun.Instance.GetLogDataCost(), Is.EqualTo(8));
+        Assert.That(Eip7667Spec.Instance.GetLogDataCost(), Is.EqualTo(10));
+    }
+
+    [Test]
+    public void GetSha256PrecompileBaseCost()
+    {
+        Assert.That(Cancun.Instance.GetSha256PrecompileBaseCost(), Is.EqualTo(60));
+        Assert.That(Eip7667Spec.Instance.GetSha256PrecompileBaseCost(), Is.EqualTo(300));
+    }
+
+    [Test]
+    public void GetSha256PrecompileWordCost()
+    {
+        Assert.That(Cancun.Instance.GetSha256PrecompileWordCost(), Is.EqualTo(12));
+        Assert.That(Eip7667Spec.Instance.GetSha256PrecompileWordCost(), Is.EqualTo(60));
+    }
+
+    [Test]
+    public void GetBlake2GFRoundDataCost()
+    {
+        Assert.That(Cancun.Instance.GetBlake2GFRoundDataCost(), Is.EqualTo(1));
+        Assert.That(Eip7667Spec.Instance.GetBlake2GFRoundDataCost(), Is.EqualTo(10));
+    }
+
+    [Test]
+    public void Keccak256OpDifference()
+    {
+        byte[] code = Prepare.EvmCode
+            .PushData(32)
+            .PushData(0)
+            .Op(Instruction.KECCAK256)
+            .STOP()
+            .Done;
+
+        TestAllTracerWithOutput resultEipDisabled = Execute((ForkActivation)1, code);
+        Setup();
+        TestAllTracerWithOutput resultEipEnabled = Execute((ForkActivation)2, code);
+
+        long gasDifference = Eip7667Spec.Instance.GetSha3Cost() - Cancun.Instance.GetSha3Cost()
+                             + Eip7667Spec.Instance.GetSha3WordCost() - Cancun.Instance.GetSha3WordCost();
+
+        Assert.That(resultEipEnabled.GasSpent - resultEipDisabled.GasSpent, Is.EqualTo(gasDifference));
```

Comment:
just have a look at how other test works regarding checking values before and after. implement this similarly. EIP3651Tests can be a good reference for that.